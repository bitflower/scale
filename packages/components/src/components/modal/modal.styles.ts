import { JssStyle } from 'jss';

export const styles: JssStyle = {
  modal: {
    zIndex: 100,
    display: 'none',
    position: 'fixed',
    top: 0,
    left: 0,
    height: 'calc(100% - 160px)',
    width: 'calc(100% - 62.5%)',
    margin: '80px 31.25%',
    alignItems: 'center',
  },
  'modal--opened': {
    display: 'flex',
  },
  'modal__scroll-container': {
    overflow: 'auto',
  },
  modal__content: {
    background: '#FFFFFF',
    color: '#333',
    borderRadius: '3px',
    boxShadow: '0 2px 8px rgba(0, 0, 0, 0.26)',
    maxHeight: '100%',
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    zIndex: 100,
  },
  modal__header: {
    fontSize: '1.25rem',
    color: '#262626',
    margin: '0 1rem',
    padding: '1rem 0',
    lineHeight: '16px',
    display: 'flex',
    justifyContent: 'space-between',
    '& ::slotted(*)': {
      fontSize: '1rem',
      color: '#262626',
      margin: 0,
      fontWeight: 'bold',
    },
  },
  'modal__header-scroll': {
    borderBottom: '1px solid #CCC',
  },
  modal__body: {
    padding: '1rem',
    lineHeight: '1',
  },
  modal__close: {
    cursor: 'pointer',
  },
  modal__actions: {
    display: 'flex',
    justifyContent: 'flex-end',
    padding: '1rem',
    background: '#FFF',
    borderRadius: '0 0 3px 3px',
  },

  modal__backdrop: {
    position: 'fixed',
    top: 0,
    left: 0,
    width: '100%',
    height: '100vh',
    background: 'rgba(0, 0, 0, 0.75)',
  },
};
