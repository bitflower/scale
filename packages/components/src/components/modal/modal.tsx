import '@proyecto26/animatable-component';
import { KEYFRAMES } from '@proyecto26/animatable-component';
import {
  Component,
  Prop,
  h,
  Element,
  Host,
  Watch,
  Event,
  EventEmitter,
} from '@stencil/core';
import { HTMLStencilElement } from '@stencil/core/internal';
import { combineObjects, CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './modal.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';
import { getTheme } from '../../theme/theme';

@Component({
  tag: 'scale-modal',
  shadow: true,
})
export class Modal implements Base {
  hasSlotHeader: boolean;
  hasSlotClose: boolean;
  hasSlotActions: boolean;
  combinedTransitions: any;

  @Element() hostElement: HTMLStencilElement;
  /** (optional) Transition overrides */
  @Prop() transitions?: any;
  /** (optional) Modal class */
  @Prop() customClass?: string = '';
  /** (optional) Modal size */
  @Prop() size?: string = 'default';
  /** (optional) Modal variant */
  @Prop() variant?: string = '';
  /** (optional) If true, the Modal is open. */
  @Prop() opened?: boolean = false;

  /** (optional) Injected jss styles */
  @Prop() styles?: StyleSheet;
  /** decorator Jss stylesheet */
  @CssInJs('Modal', styles) stylesheet: StyleSheet;

  /** (optional) Callback fired when the component requests to be closed. */
  @Event() scaleClose?: EventEmitter<MouseEvent>;

  constructor() {
    this.close = this.close.bind(this);
    this.animateComponent = this.animateComponent.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
  }

  @Watch('opened')
  async watchHandler() {
    await this.animateComponent();
  }

  async close(event?: MouseEvent) {
    this.scaleClose.emit(event);
  }

  waitForChildren(children) {
    return new Promise(resolve => {
      const findChildren = () =>
        children.length
          ? resolve()
          : setTimeout(() => {
              findChildren();
            });

      findChildren();
    });
  }

  decorateScrollContainer() {
    const scrollContainer = this.hostElement.shadowRoot.querySelector(
      `.${this.stylesheet.classes['modal__scroll-container']}`
    );
    const modalHeader = this.hostElement.shadowRoot.querySelector(
      `.${this.stylesheet.classes.modal__header}`
    );
    setTimeout(() => {
      const hasVerticalScrollbar =
        scrollContainer.scrollHeight > scrollContainer.clientHeight;

      if (hasVerticalScrollbar) {
        modalHeader.classList.add(
          this.stylesheet.classes['modal__header-scroll']
        );
      }
    });
  }

  async animateComponent() {
    const direction = this.opened ? 'IN' : 'OUT';

    if (!this.hostElement.shadowRoot) {
      // in this case we are doing SSR, so we can skip animations
      return null;
    }
    await this.waitForChildren(this.hostElement.shadowRoot.children);

    this.decorateScrollContainer();

    let combinedTransitions;
    try {
      combinedTransitions = JSON.parse(this.transitions);
    } catch (err) {
      combinedTransitions = this.transitions;
    }
    combinedTransitions = combineObjects(
      getTheme().components.Modal.transitions,
      combinedTransitions
    );

    const { backDrop, modalContent } = combinedTransitions;
    const { transition: transitionModal, ...optionsModal } = modalContent[
      direction
    ];
    const { transition: transitionBackDrop, ...optionsBackDrop } = backDrop[
      direction
    ];

    const animationModal = this.hostElement.shadowRoot
      .querySelector(`.${this.stylesheet.classes.modal__content}`)
      .animate(KEYFRAMES[transitionModal], optionsModal);

    const animationBackdrop = this.hostElement.shadowRoot
      .querySelector(`.${this.stylesheet.classes.modal__backdrop}`)
      .animate(KEYFRAMES[transitionBackDrop], optionsBackDrop);

    const modalClassList = this.hostElement.shadowRoot.querySelector(
      `.${this.stylesheet.classes.modal}`
    ).classList;

    if (direction === 'IN') {
      modalClassList.add(this.stylesheet.classes['modal--opened']);
      document.addEventListener('keydown', this.handleKeyDown);
    }

    animationBackdrop.play();
    animationModal.play();

    return new Promise(resolve => {
      animationModal.onfinish = function() {
        if (direction === 'OUT') {
          modalClassList.remove(this.stylesheet.classes['modal--opened']);
          document.removeEventListener('keydown', this.handleKeyDown);
        }
        resolve();
      }.bind(this);
    });
  }

  componentWillLoad() {
    this.hasSlotHeader = !!this.hostElement.querySelector('[slot="header"]');
    this.hasSlotClose = !!this.hostElement.querySelector('[slot="close"]');
    this.hasSlotActions = !!this.hostElement.querySelector(
      '[slot="modal-actions"]'
    );
  }
  componentWillUpdate() {}
  componentDidUnload() {}
  handleKeyDown(event) {
    if (event.key === 'Escape') {
      this.close(event);
    }
  }
  componentDidLoad() {
    this.animateComponent();
  }

  render() {
    const { classes } = this.stylesheet;

    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <animatable-component>
          <div class={this.getCssClassMap()}>
            <div class={classes.modal__backdrop} onClick={this.close}></div>

            <div class={classes.modal__content}>
              {this.hasSlotHeader /* istanbul ignore next */ && (
                <div class={classes.modal__header}>
                  <slot name="header" />
                  <a class={classes.modal__close} onClick={this.close}>
                    {this.hasSlotClose ? (
                      <div class={classes['modal__close-icon']}>
                        <slot name="close" />
                      </div>
                    ) : (
                      'x'
                    )}
                  </a>
                </div>
              )}

              <div class={classes['modal__scroll-container']}>
                <div class={classes.modal__body}>
                  <slot />
                </div>
              </div>

              <div class={classes.modal__actions}>
                <slot name="modal-actions" />
              </div>
            </div>
          </div>
        </animatable-component>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes.modal,
      this.customClass && this.customClass,
      this.size && classes[`modal--size-${this.size}`],
      this.variant && classes[`modal--variant-${this.variant}`]
    );
  }
}
