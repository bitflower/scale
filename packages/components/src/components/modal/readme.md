# scale-modal



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                            | Type                                     | Default     |
| ------------- | -------------- | -------------------------------------- | ---------------------------------------- | ----------- |
| `customClass` | `custom-class` | (optional) Modal class                 | `string`                                 | `''`        |
| `opened`      | `opened`       | (optional) If true, the Modal is open. | `boolean`                                | `false`     |
| `size`        | `size`         | (optional) Modal size                  | `string`                                 | `'default'` |
| `styles`      | --             | (optional) Injected jss styles         | `StyleSheet<string \| number \| symbol>` | `undefined` |
| `transitions` | `transitions`  | (optional) Transition overrides        | `any`                                    | `undefined` |
| `variant`     | `variant`      | (optional) Modal variant               | `string`                                 | `''`        |


## Events

| Event        | Description                                                         | Type                      |
| ------------ | ------------------------------------------------------------------- | ------------------------- |
| `scaleClose` | (optional) Callback fired when the component requests to be closed. | `CustomEvent<MouseEvent>` |


## Dependencies

### Depends on

- animatable-component

### Graph
```mermaid
graph TD;
  scale-modal --> animatable-component
  style scale-modal fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
