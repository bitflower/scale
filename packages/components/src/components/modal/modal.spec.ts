import { newSpecPage } from '@stencil/core/testing';
import { Modal } from './modal';
import { styles } from './modal.styles';
import jss from 'jss';

describe('Modal', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Modal();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  const components = [Modal];

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Modal],
      html: `<scale-modal>Label</scale-modal>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot with header slot', async () => {
    const page = await newSpecPage({
      components: [Modal],
      html: `
			<scale-modal>
				<span slot="header">Header content</span>
				A title
			</scale-modal>
			`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot with actions slot', async () => {
    const page = await newSpecPage({
      components: [Modal],
      html: `
			<scale-modal>
				<span slot="modal-actions">Action buttons</span>
				Content
			</scale-modal>
			`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot with close slot', async () => {
    const page = await newSpecPage({
      components: [Modal],
      html: `
			<scale-modal>
				<span slot="close">Close</span>
				Content
			</scale-modal>
			`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should match snapshot when opened', async () => {
    const page = await newSpecPage({
      components,
      html: `<scale-modal opened=true>Label</scale-modal>`,
    });
    expect(page.root.shadowRoot).toBeTruthy();
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');

    element.size = 'small';
    stylesheet.addRule('modal--size-small', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['modal--size-small']
    );

    element.variant = 'primary';
    stylesheet.addRule('modal--variant-primary', {});
    expect(element.getCssClassMap()).toContain(
      stylesheet.classes['modal--variant-primary']
    );
  });
});
