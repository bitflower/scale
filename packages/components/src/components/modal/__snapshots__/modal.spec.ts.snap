// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `
<scale-modal>
  <mock:shadow-root>
    <style>
      .modal-0-0-11 {
  top: 0;
  left: 0;
  width: calc(100% - 62.5%);
  height: calc(100% - 160px);
  margin: 80px 31.25%;
  display: none;
  z-index: 100;
  position: fixed;
  align-items: center;
}
.modal--opened-0-0-12 {
  display: flex;
}
.modal__scroll-container-0-0-13 {
  overflow: auto;
}
.modal__content-0-0-14 {
  color: #333;
  width: 100%;
  display: flex;
  z-index: 100;
  background: #FFFFFF;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.26);
  max-height: 100%;
  border-radius: 3px;
  flex-direction: column;
  justify-content: space-between;
}
.modal__header-0-0-15 {
  color: #262626;
  margin: 0 1rem;
  display: flex;
  padding: 1rem 0;
  font-size: 1.25rem;
  line-height: 16px;
  justify-content: space-between;
}
.modal__header-0-0-15 ::slotted(*) {
  color: #262626;
  margin: 0;
  font-size: 1rem;
  font-weight: bold;
}
.modal__header-scroll-0-0-16 {
  border-bottom: 1px solid #CCC;
}
.modal__body-0-0-17 {
  padding: 1rem;
  line-height: 1;
}
.modal__close-0-0-18 {
  cursor: pointer;
}
.modal__actions-0-0-19 {
  display: flex;
  padding: 1rem;
  background: #FFF;
  border-radius: 3px;
  justify-content: flex-end;
}
.modal__backdrop-0-0-20 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  position: fixed;
  background: rgba(0, 0, 0, 0.75);
}
.transitions-0-0-21 {}
    </style>
    <animatable-component>
      <div class="modal-0-0-11">
        <div class="modal__backdrop-0-0-20"></div>
        <div class="modal__content-0-0-14">
          <div class="modal__scroll-container-0-0-13">
            <div class="modal__body-0-0-17">
              <slot></slot>
            </div>
          </div>
          <div class="modal__actions-0-0-19">
            <slot name="modal-actions"></slot>
          </div>
        </div>
      </div>
    </animatable-component>
  </mock:shadow-root>
  Label
</scale-modal>
`;

exports[`Modal should match snapshot when opened 1`] = `
<scale-modal opened="true">
  <mock:shadow-root>
    <style>
      .modal-0-0-11 {
  top: 0;
  left: 0;
  width: calc(100% - 62.5%);
  height: calc(100% - 160px);
  margin: 80px 31.25%;
  display: none;
  z-index: 100;
  position: fixed;
  align-items: center;
}
.modal--opened-0-0-12 {
  display: flex;
}
.modal__scroll-container-0-0-13 {
  overflow: auto;
}
.modal__content-0-0-14 {
  color: #333;
  width: 100%;
  display: flex;
  z-index: 100;
  background: #FFFFFF;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.26);
  max-height: 100%;
  border-radius: 3px;
  flex-direction: column;
  justify-content: space-between;
}
.modal__header-0-0-15 {
  color: #262626;
  margin: 0 1rem;
  display: flex;
  padding: 1rem 0;
  font-size: 1.25rem;
  line-height: 16px;
  justify-content: space-between;
}
.modal__header-0-0-15 ::slotted(*) {
  color: #262626;
  margin: 0;
  font-size: 1rem;
  font-weight: bold;
}
.modal__header-scroll-0-0-16 {
  border-bottom: 1px solid #CCC;
}
.modal__body-0-0-17 {
  padding: 1rem;
  line-height: 1;
}
.modal__close-0-0-18 {
  cursor: pointer;
}
.modal__actions-0-0-19 {
  display: flex;
  padding: 1rem;
  background: #FFF;
  border-radius: 3px;
  justify-content: flex-end;
}
.modal__backdrop-0-0-20 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  position: fixed;
  background: rgba(0, 0, 0, 0.75);
}
.transitions-0-0-21 {}
    </style>
    <animatable-component>
      <div class="modal-0-0-11">
        <div class="modal__backdrop-0-0-20"></div>
        <div class="modal__content-0-0-14">
          <div class="modal__scroll-container-0-0-13">
            <div class="modal__body-0-0-17">
              <slot></slot>
            </div>
          </div>
          <div class="modal__actions-0-0-19">
            <slot name="modal-actions"></slot>
          </div>
        </div>
      </div>
    </animatable-component>
  </mock:shadow-root>
  Label
</scale-modal>
`;

exports[`Modal should match snapshot with actions slot 1`] = `
<scale-modal>
  <mock:shadow-root>
    <style>
      .modal-0-0-53 {
  top: 0;
  left: 0;
  width: calc(100% - 62.5%);
  height: calc(100% - 160px);
  margin: 80px 31.25%;
  display: none;
  z-index: 100;
  position: fixed;
  align-items: center;
}
.modal--opened-0-0-54 {
  display: flex;
}
.modal__scroll-container-0-0-55 {
  overflow: auto;
}
.modal__content-0-0-56 {
  color: #333;
  width: 100%;
  display: flex;
  z-index: 100;
  background: #FFFFFF;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.26);
  max-height: 100%;
  border-radius: 3px;
  flex-direction: column;
  justify-content: space-between;
}
.modal__header-0-0-57 {
  color: #262626;
  margin: 0 1rem;
  display: flex;
  padding: 1rem 0;
  font-size: 1.25rem;
  line-height: 16px;
  justify-content: space-between;
}
.modal__header-0-0-57 ::slotted(*) {
  color: #262626;
  margin: 0;
  font-size: 1rem;
  font-weight: bold;
}
.modal__header-scroll-0-0-58 {
  border-bottom: 1px solid #CCC;
}
.modal__body-0-0-59 {
  padding: 1rem;
  line-height: 1;
}
.modal__close-0-0-60 {
  cursor: pointer;
}
.modal__actions-0-0-61 {
  display: flex;
  padding: 1rem;
  background: #FFF;
  border-radius: 3px;
  justify-content: flex-end;
}
.modal__backdrop-0-0-62 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  position: fixed;
  background: rgba(0, 0, 0, 0.75);
}
.transitions-0-0-63 {}
    </style>
    <animatable-component>
      <div class="modal-0-0-53">
        <div class="modal__backdrop-0-0-62"></div>
        <div class="modal__content-0-0-56">
          <div class="modal__scroll-container-0-0-55">
            <div class="modal__body-0-0-59">
              <slot></slot>
            </div>
          </div>
          <div class="modal__actions-0-0-61">
            <slot name="modal-actions"></slot>
          </div>
        </div>
      </div>
    </animatable-component>
  </mock:shadow-root>
  <span slot="modal-actions">
    Action buttons
  </span>
  Content
</scale-modal>
`;

exports[`Modal should match snapshot with close slot 1`] = `
<scale-modal>
  <mock:shadow-root>
    <style>
      .modal-0-0-74 {
  top: 0;
  left: 0;
  width: calc(100% - 62.5%);
  height: calc(100% - 160px);
  margin: 80px 31.25%;
  display: none;
  z-index: 100;
  position: fixed;
  align-items: center;
}
.modal--opened-0-0-75 {
  display: flex;
}
.modal__scroll-container-0-0-76 {
  overflow: auto;
}
.modal__content-0-0-77 {
  color: #333;
  width: 100%;
  display: flex;
  z-index: 100;
  background: #FFFFFF;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.26);
  max-height: 100%;
  border-radius: 3px;
  flex-direction: column;
  justify-content: space-between;
}
.modal__header-0-0-78 {
  color: #262626;
  margin: 0 1rem;
  display: flex;
  padding: 1rem 0;
  font-size: 1.25rem;
  line-height: 16px;
  justify-content: space-between;
}
.modal__header-0-0-78 ::slotted(*) {
  color: #262626;
  margin: 0;
  font-size: 1rem;
  font-weight: bold;
}
.modal__header-scroll-0-0-79 {
  border-bottom: 1px solid #CCC;
}
.modal__body-0-0-80 {
  padding: 1rem;
  line-height: 1;
}
.modal__close-0-0-81 {
  cursor: pointer;
}
.modal__actions-0-0-82 {
  display: flex;
  padding: 1rem;
  background: #FFF;
  border-radius: 3px;
  justify-content: flex-end;
}
.modal__backdrop-0-0-83 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  position: fixed;
  background: rgba(0, 0, 0, 0.75);
}
.transitions-0-0-84 {}
    </style>
    <animatable-component>
      <div class="modal-0-0-74">
        <div class="modal__backdrop-0-0-83"></div>
        <div class="modal__content-0-0-77">
          <div class="modal__scroll-container-0-0-76">
            <div class="modal__body-0-0-80">
              <slot></slot>
            </div>
          </div>
          <div class="modal__actions-0-0-82">
            <slot name="modal-actions"></slot>
          </div>
        </div>
      </div>
    </animatable-component>
  </mock:shadow-root>
  <span slot="close">
    Close
  </span>
  Content
</scale-modal>
`;

exports[`Modal should match snapshot with header slot 1`] = `
<scale-modal>
  <mock:shadow-root>
    <style>
      .modal-0-0-32 {
  top: 0;
  left: 0;
  width: calc(100% - 62.5%);
  height: calc(100% - 160px);
  margin: 80px 31.25%;
  display: none;
  z-index: 100;
  position: fixed;
  align-items: center;
}
.modal--opened-0-0-33 {
  display: flex;
}
.modal__scroll-container-0-0-34 {
  overflow: auto;
}
.modal__content-0-0-35 {
  color: #333;
  width: 100%;
  display: flex;
  z-index: 100;
  background: #FFFFFF;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.26);
  max-height: 100%;
  border-radius: 3px;
  flex-direction: column;
  justify-content: space-between;
}
.modal__header-0-0-36 {
  color: #262626;
  margin: 0 1rem;
  display: flex;
  padding: 1rem 0;
  font-size: 1.25rem;
  line-height: 16px;
  justify-content: space-between;
}
.modal__header-0-0-36 ::slotted(*) {
  color: #262626;
  margin: 0;
  font-size: 1rem;
  font-weight: bold;
}
.modal__header-scroll-0-0-37 {
  border-bottom: 1px solid #CCC;
}
.modal__body-0-0-38 {
  padding: 1rem;
  line-height: 1;
}
.modal__close-0-0-39 {
  cursor: pointer;
}
.modal__actions-0-0-40 {
  display: flex;
  padding: 1rem;
  background: #FFF;
  border-radius: 3px;
  justify-content: flex-end;
}
.modal__backdrop-0-0-41 {
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  position: fixed;
  background: rgba(0, 0, 0, 0.75);
}
.transitions-0-0-42 {}
    </style>
    <animatable-component>
      <div class="modal-0-0-32">
        <div class="modal__backdrop-0-0-41"></div>
        <div class="modal__content-0-0-35">
          <div class="modal__header-0-0-36">
            <slot name="header"></slot>
            <a class="modal__close-0-0-39">
              x
            </a>
          </div>
          <div class="modal__scroll-container-0-0-34">
            <div class="modal__body-0-0-38">
              <slot></slot>
            </div>
          </div>
          <div class="modal__actions-0-0-40">
            <slot name="modal-actions"></slot>
          </div>
        </div>
      </div>
    </animatable-component>
  </mock:shadow-root>
  <span slot="header">
    Header content
  </span>
  A title
</scale-modal>
`;
